### ISSUE TITLE
Print the list of transactions

### ISSUE THREAD

**Issue by rwobben** [rwobben/paytime#29]
Do you mean here print a list of transactions and let it look like we did on "show accounts"

I think I also will like it as we do this : 

Accountname       accountBalance     Amount   Description 

test1                        200                         100        Buyed stamps 

So more a sort of table layout. 

When we have done this, I also like to see it on Show Accounts. 

Good idea ??

**Comment by GetContented** [comment]
If you think it's a good idea, sure.

**Comment by rwobben** [comment]
I think we can begin with the "show account part" 

That one schould look like this : 

Name            Amount 

So first think how big name and account needs to be. 
I was thinking to convert both to a fixed length string so everything looks nice.

**Comment by GetContented** [comment]
Ok. The Data.Text.Format module has functions left and right that can do fixed-length padding, however they work with Text rather than String.

When we start getting serious with our application, String won't cut it for textual representation. It's just a simple [Char] type, after all. That means it's not the best for a whole bunch of reasons.

When we move the application into a web server, we'll need to deal with using Text or ByteString then, too... but this is potentially much more of a headache than you anticipate for us at the moment.

I guess the question is... is it worth it, given that we're going to be moving the application to a web server at some stage.

**Comment by rwobben** [comment]
I think for me it's a exercise in recursion and how to get data out of a map.
I know we can just print out the map but I find this no challenge.

I found this code so we can make a fixed length string.

separator :: String -> String 
separator s = replicate (lineLength - length s) '.' 

 
formatLine :: (Name,Price) -> String 
 formatLine (name, pence) = name ++ (separator name) ++ (formatPence pence) ++ "\n" 

We can use this as temporarily solution. 

Later when we convert it to Text we could make a better solution and delete this one. 
For me , this is no problem.

**Comment by GetContented** [comment]
Yeah, that simple solution might work fine for now. You need to consider the edge cases, though... what should happen when it's too big as well as when it's too small?

Cool, so are you going to try to tackle this one on your own? What's your plan of attack? How will you decompose this problem into smaller pieces?

You can start with what you know (which seems to be what you're already doing... you know you'll need a function to take some data, and format it into fixed-width sections.). This is called bottom-up design. You can tackle the other end at the same time, splitting the big task into several smaller ones, if they're obvious to you.

**Comment by rwobben** [comment]
I can try to do this on my own. 

I think I will ttry to do this in 3 bigger steps.

1) I think the lib.hs is becoming to big. So I will try to make 3 seperate libs for accounts. transactions and reports. 

2)  Make some changes so we know if the transaction is a withdrawl or a deposit. 

3) Make the report. 
-  Make it work so descriptiuon has a fixed length. So we need a check for the length. 
-  Make it work for the withdrawl and deposit 
-  Make it work so we can see a sub-amount after every transaction. 

Maybe on the way I can make the steps to smaller steps but I see on when working.

**Comment by GetContented** [comment]
If you want to pull modules out, it's a good idea to do this separately than while you're working on new code.

You should delineate modules by their naturally occurring lines rather than some artificial idea you have of where they should break.

**Comment by rwobben** [comment]
Of Course. that's why I do make this the first step. No no new code is introduced. 
The only thing I try to rewrite is the menu.  I like to make seperate functions for every submenu. 
See bug #30

**Comment by GetContented** [comment]
Ok :) hehe...
